src/lib.rs

```diff
@@ -120,3 +120,65 @@ impl DocSource for AggregateDocSource {
         unimplemented!();
     }
 }
+
+#[repr(transparent)]
+pub struct Lowercase<'a>(pub &'a [u8]);
+
+pub(crate) fn starts_with_insensitive_ascii(s: &[u8], prefix: &Lowercase) -> bool {
+    let prefix = prefix.0;
+
+    if s.len() < prefix.len() {
+        return false;
+    }
+
+    for (i, b) in prefix.into_iter().enumerate() {
+        if unsafe { s.get_unchecked(i) }.to_ascii_lowercase() != *b {
```

Comment:
you should definitely give a thorough explaination of _why_ this unsafe is _always_ safe, to make sure noone accidentally breaks that contract in the future